### ISSUE TITLE
LDAP Username instead of GUID

### ISSUE THREAD

**Issue by jdhirst1** [nextcloud/social#317]
**Describe the bug**
When using an Active Directory LDAP backend, NextCloud social shows user as "objectGuid@server.tld". This works, however it is quite unweildy as nobody wants to be known as @5DD3ED74-0B77-4501-B8DE-D3F5687B6F99@server.tld.

**To Reproduce**
Enable LDAP authentication with Active Directory, and install and enable social app. All user ids including federated sharing are based upon objectGUID rather than the samAccountName or other configurable value.

**Expected behavior**
The username used to login should be used to create the social ID rather than the objectGuid from LDAP.

<details>
<summary>Server details</summary>

**Social app version:**
0.1.2

**Operating system**:
CentOS 7.6.1810
**Web server:**
Nginx 1.14.1-centos7.18120419
**Database:**
MySQL Community 5.5.60-1.el7_5
**PHP version:**
7.2.13
**Nextcloud version:** (see Nextcloud admin page)
15.0.0
</details>

**Comment by daita** [comment]
@juliushaertl @jancborchardt  what about this ? when you asked to force the social account based on current nextcloud account, you were saying that this can be fixed within the LDAP setup ?

**Comment by jancborchardt** [comment]
@jdhirst1 we generally aim to have as few settings and configuration options as possible to make it just work. Is it possible for example to:

- Detect if this LDAP name mapping has been set up
- If it hasn’t, don’t automatically set up Nextcloud Social and ask the admin to set up this domain mapping or accept that all the user identifiers are long character strings
- If it has, just do our normal thing and people will have normal handles.

**Comment by daita** [comment]
I have almost no knowledge in LDAP, but the code of the Social app allow to define a different username than the userid when creating a social account. We just need to define a way to link a user to the wanted social-username in a configurable way.

**Comment by phaus** [comment]
I had the same problem.
You need to go to the LDAP Settings and into Expert mode.
Then you can override the default UUIDs:

<img width="1307" alt="screenshot 2019-02-04 05 36 29" src="https://user-images.githubusercontent.com/346361/52190350-f1b89500-283e-11e9-9668-c6ed3f327128.png">

Unfortunetally, you will loose the connection between user and data. So you will get new users and you have to copy the files manually to the specific user folder afterwards.

**Comment by daita** [comment]
@phaus thanks for your answer, would it be possible to assign the username to a custom variable and get it from the front-end ?

**Comment by MorrisJobke** [comment]
> @phaus thanks for your answer, would it be possible to assign the username to a custom variable and get it from the front-end ?

There is no such option in Nextcloud. Keep in mind that the userID could always be super cryptic. This is the reason why we have also a login name (which is by default the email address if it is unique) and the display name. There is no way around this.

**Comment by MorrisJobke** [comment]
cc @schiessle @nickvergessen @rullzer Same stuff that is also with the federation ID present: the userID is the internal representation and it was a bad habit to use it in user facing parts of our system. It could be cryptic, long and only works by accident to login. We really need to think about this a bit more. :(

**Comment by blizzz** [comment]
In federated sharing there is a similar problem, and a – given ugly – workaround to use and resolved the (last) loginname. It's stored in the user session. This also means it is (without further magic, there's nothing in place to reverse lookup a login attribute) only known after a first login.

**Comment by wesleylc1** [comment]
Hi guys.
My nextcloud backend is not syncing the "ou = user" userbase, but workstation import is done "ou = stations".

![Captura de tela de 2019-09-10 15-17-07](https://user-images.githubusercontent.com/29066796/64639439-3a8c8980-d3de-11e9-9535-58d967994603.png)

**Comment by victort** [comment]
hi, just adding my `US$0.02++`, could it just be as simple as
```
if using ldap
  set username to %uid
```
or something to that effect?

**Comment by wesleylc1** [comment]
@victort Hi man, how are you?
Really worked after setting user to use% uid
![Captura de tela de 2019-10-04 15-54-37](https://user-images.githubusercontent.com/29066796/66232881-4b1fdf00-e6c0-11e9-9a07-34f47c6a2997.png)

**Comment by victort** [comment]
yeah. then i have to redo my whole setup, when all the uuids are wiped and suddenly none of the files or calendars map anymore.

also, i'd like (to figure out how) to get it to not say `@localhost` while i'm at it.

**Comment by victort** [comment]
yay, figured it out. (it was the `overwrite.cli.url` variable in `config.php`)

**Comment by SchBenedikt** [comment]
Automatically closed — LDAP username handling has been improved. The account service now properly resolves display names instead of showing internal usernames.
